From 301ad592142079d36f4d33f5309c103c7f4f5dfb Mon Sep 17 00:00:00 2001
From: Gilles <erans@apache.org>
Date: Wed, 12 Nov 2014 12:16:13 +0100
Subject: [PATCH] MATH-1167

Use user-specified threshold to allow detection of singular matrix.
[Reported, with fix, by Neil Ireson.]
---
 src/changes/changes.xml                                       | 4 ++++
 .../math3/stat/regression/OLSMultipleLinearRegression.java    | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/changes/changes.xml b/src/changes/changes.xml
index df71204e58..5377621129 100644
--- a/src/changes/changes.xml
+++ b/src/changes/changes.xml
@@ -73,6 +73,10 @@ Users are encouraged to upgrade to this version as this release not
   2. A few methods in the FastMath class are in fact slower that their
   counterpart in either Math or StrictMath (cf. MATH-740 and MATH-901).
 ">
+      <action dev="erans" type="fix" issue="MATH-1167" due-to="Neil Ireson">
+        "o.a.c.m.stat.regression.OLSMultipleLinearRegression": Use threshold
+        when performing "QRDecomposition".
+      </action>
       <action dev="tn" type="fix" issue="MATH-1165" due-to="Pashutan Modaresi">
         "FuzzyKMeansClusterer" has thrown an exception in case one of the data
         points was equal to a cluster center.
diff --git a/src/main/java/org/apache/commons/math3/stat/regression/OLSMultipleLinearRegression.java b/src/main/java/org/apache/commons/math3/stat/regression/OLSMultipleLinearRegression.java
index ba86c5de9d..8c5dc4c958 100644
--- a/src/main/java/org/apache/commons/math3/stat/regression/OLSMultipleLinearRegression.java
+++ b/src/main/java/org/apache/commons/math3/stat/regression/OLSMultipleLinearRegression.java
@@ -233,7 +233,7 @@ public double calculateAdjustedRSquared() throws MathIllegalArgumentException {
     @Override
     protected void newXSampleData(double[][] x) {
         super.newXSampleData(x);
-        qr = new QRDecomposition(getX());
+        qr = new QRDecomposition(getX(), threshold);
     }
 
     /**
